bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccAPI.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.evcc.internal;
+
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.EVCC_REST_API;
+import static org.openhab.binding.evcc.internal.EvccBindingConstants.LONG_CONNECTION_TIMEOUT_MILLISEC;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.evcc.internal.dto.Result;
+import org.openhab.binding.evcc.internal.dto.Status;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link EvccAPI} is responsible for API calls to evcc.
+ * 
+ * @author Florian Hotze - Initial contribution
+ */
+public class EvccAPI {
+    private static final Logger logger = LoggerFactory.getLogger(EvccAPI.class);
+    private static final Gson gson = new Gson();
+    private final @Nullable String host;
+
+    public EvccAPI(@Nullable String host) {
```

Comment:
Does it make sense to instantiate without a host, since it won't be able to do anything? The `@Nullable` annotation could be removed, and caller will need to handle configuration problems etc.